src/autocomplete.mjs

```diff
@@ -230,24 +241,30 @@ export default class extends Controller {
   }
 
   open() {
-    if (!this.resultsTarget.hidden) return
-    this.resultsTarget.hidden = false
+    if (!this.isHidden) return
+    if (!this.hasSkipHiddenPropertyValue) {
+      this.resultsTarget.hidden = false
+    }
+    this.isHidden = false;
     this.element.setAttribute("aria-expanded", "true")
     this.element.dispatchEvent(
       new CustomEvent("toggle", {
-        detail: { input: this.input, results: this.results }
+        detail: { action: 'open', input: this.input, results: this.results }
```

Comment:
Unless I'm missing something, `this.input` and `this.results` are undefined 100% of the time... and it looks like it was always this way.

If the intention was for these to be `this.inputTarget` and `this.resultsTarget`, we could change those. But I wanted to keep my PR diff as small as possible :).

Reply:
That's fair. You have my blessing in broadly fix external events since you're looking at them/care about them. This PR or another is good by me.

Reply:
Beautiful then :). I've updated these to be the two targets, which I think could be useful :).